shortfin/python/shortfin_apps/llm/components/debug_service.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+Some code for debugging service.py. Importing this should do nothing and import no additional dependencies if DEBUG_LLM_SERVICE == False
+"""
+
+
+import asyncio
+import logging
+from datetime import datetime
+from pathlib import Path
+import json
+import numpy as np
+import pandas as pd
+from typing import List, Dict, Any
+from pprint import pformat
+
+logger = logging.getLogger(__name__)
+
+import os
+
+# Get environment variable, default to False if not set
+SHORTFIN_DEBUG_LLM_SERVICE = os.getenv(
+    "SHORTFIN_DEBUG_LLM_SERVICE", "False"
+).lower() in ("true", "yes", "1", "y")
+if SHORTFIN_DEBUG_LLM_SERVICE:
+    logger.info("DEBUG_LLM_SERVICE=True")
+    dump_id = 0
+    boot_timestamp = datetime.now().isoformat()
+    DEBUG_DATA_DIR = Path.home() / ".shortfin/debug/"
+    DUMP_DIR_THIS_SESSION = (
+        DEBUG_DATA_DIR / "llm_service_invocation_dumps" / "{boot_timestamp}"
+    )
+    DUMP_DIR_THIS_SESSION.mkdir(parents=True, exist_ok=False)
+    logger.info(
+        f"[debug_service.py] Please find debug dumps for service.py in {DUMP_DIR_THIS_SESSION}"
+    )
```

Comment:
Can you move this to an `init()` function so it doesn't always run at import time? This debug file could provide a class instead of a loose function that uses global variables. Then there would be no need to duplicate this `os.getenv` code across both files or add the `Importing this should do nothing` comment

```python
# debug_service.py

class ShortfinLLMDebugService:
  def __init__(self):
    dump_id = 0
    dump_dir = ""
    dump_dir.mkdir(...)

  async def pre_invocation_debug_dump(...):
    is_decode = ...

# service.py

class GenerateService:
    def __init__(self, ...):
      ...

      SHORTFIN_DEBUG_LLM_SERVICE = os.getenv(
          "SHORTFIN_DEBUG_LLM_SERVICE", "False"
      ).lower() in ("true", "yes", "1", "y")
      if SHORTFIN_DEBUG_LLM_SERVICE:
        from .debug_service import ShortfinLLMDebugService
        llm_debug_service = ShortfinLLMDebugService

...

    async def run(self):
      ...
      if self.llm_debug_service:
        llm_debug_service.pre_invocation_debug_dump(...)
```